src/IECoreScene/MeshAlgoDistributePoints.cpp

```diff
@@ -207,6 +207,14 @@ PointsPrimitivePtr MeshAlgo::distributePoints( const MeshPrimitive *mesh, float
 	MeshPrimitiveEvaluatorPtr meshEvaluator = new MeshPrimitiveEvaluator( updatedMesh );
 
 	ConstV2fVectorDataPtr uvData = updatedMesh->expandedVariableData<V2fVectorData>( uvSet, PrimitiveVariable::FaceVarying );
+
+	if ( !uvData )
+	{
+		throw InvalidArgumentException(
+			boost::str( boost::format("MeshAlgo::distributePoints : uvSet '%1%' empty or has non 'FaceVarying' interpolation.") % uvSet )
```

Comment:
What does "empty" mean in this context? I'm pretty sure `processMesh()` will error if any primitive variable has null data, so I think if we're here it's actually because the primvar doesn't exist at all, or has the wrong type (or isn't FaceVarying). Might be nice to clarify the message?

Alternatively, I wonder if we should deal with this in `expandedVariableData()`, which is the only place that knows whether the problem is non-existence, or wrong type, or wrong interpolation? Should we add a flag to that call so that it can optionally be responsible for throwing the most accurate description possible?

Reply:
> Alternatively, I wonder if we should deal with this in expandedVariableData(), which is the only place that knows whether the problem is non-existence, or wrong type, or wrong interpolation? Should we add a flag to that call so that it can optionally be responsible for throwing the most accurate description possible?

I like the idea of more informative exception messages and update the code as you suggest.